en/tutorial/nixos-install-with-disko.md

```diff
@@ -0,0 +1,176 @@
+
+### Choosing the disk configuration
+
+Disko provides a few examples to choose [from](https://github.com/nix-community/disko/tree/master/example). We will use the [hybrid](https://github.com/nix-community/disko/blob/master/example/hybrid.nix) example as it will work for both BIOS and UEFI systems.
+
+Copy the disk configuration on to the USB flash drive.
+
+```bash
+curl https://raw.githubusercontent.com/nix-community/disko/master/example/hybrid.nix -o /tmp/disko-config.nix
+```
+
+### Modify the disk configuration
+
+We need to find the device name of the disk we want to install NixOS on. We can use `lsblk` to find it.
+
+
+:::{.center}
+![[nixos-lsblk.jpeg]]
+:::
+
+In this case, the device name is `vda`. We will use this to modify `disko-config.nix` we downloaded earlier.
+
+:::{.center}
+![[nixos-disko-config.jpeg]]
+:::
+
+### Run the partitioning script
+
+>[!note]
+> The disk will be partitioned and mounted at `/mnt`.
+
+```bash
+sudo nix --experimental-features "nix-command flakes" run github:nix-community/disko -- --mode disko /tmp/disko-config.nix
+```
+
+:::{.center}
+![[nixos-disko-post-partition.jpeg]]
+:::
+
+{#configuration}
+## The NixOS configuration
+
+```bash
+sudo nixos-generate-config --no-filesystems --root /mnt
+```
+
+The [fileSystems](https://search.nixos.org/options?channel=23.11&show=fileSystems&from=0&size=50&sort=relevance&type=packages&query=fileSystems) configuration will be added by `disko`'s [nixosModule](https://nixos.wiki/wiki/NixOS_modules) hence we use `--no-filesystems` to avoid generating it. `--root` is to specify the mountpoint to generate `configuration.nix` and `hardware-configuration.nix` in. Here, it will be `/mnt/etc/nixos`.
+
+### Flakify the NixOS configuration
+
+This step remains the same as the [Install NixOS with Flake configuration on Git](https://nixos.asia/en/tutorial/nixos-install#flakeify) tutorial.
+
+### Add the `disko` nixosModule
+
+Add the `disko` flake input:
+
+```nix
+# In `flake.nix`
+{
+  inputs = {
+    disko.url = "github:nix-community/disko";
+    disko.inputs.nixpkgs.follows = "nixpkgs";
+  };
+}
+```
+>[!info]
+> `disko.inputs.nixpkgs.follows = "nixpkgs";` is to ensure that `disko` uses the same version of `nixpkgs` as specified in the current flake. This avoids having two different sources of `nixpkgs` and saves space.
+
+Add the `disko` nixosModule:
+
+```nix
+{
+  # In `outputs` of `flake.nix`
+  nixosConfigurations.nixos = {
+    # ...
+    modules = [
+      ./configuration.nix
+      disko.nixosModules.disko
+    ];
+  };
+}
+```
+:::{.center}
+![[nixos-flake-with-disko.jpeg]]
+:::
+
+Add the disk configuration and use GRUB:
+
+```nix
+{
+  # In `configuration.nix`
+  imports = [
+    ./hardware-configuration.nix
+    ./disko-config.nix
```

Comment:
Doesn't this file exist only in `/tmp/disko-config.nix`? There appears to be a step missing telling the user to move it to a suitable place.

Reply:
Ah, yes! will add that